pkgs/by-name/ge/gema/package.nix

```diff
@@ -0,0 +1,37 @@
+{
+  lib,
+  stdenv,
+  fetchgit,
+  installShellFiles,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gema";
+  version = "2.0";
+
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/gema/code.git";
```

Comment:
I think fetching source archives is preferred over a (shallow?) git clone (_note to self: look into the implementation of `fetchgit` to check its performance impact_). Would using the following release archive be an acceptable alternative, @quag?

https://sourceforge.net/projects/gema/files/gema/gema-2.0/gema-2.0-source.tar.gz/download

This would allow vor `finalAttrs.version` to be used without the need to `replaceStrings` (see below).

Reply:
The performance impact makes sense. I've switched it from the fetchgit to fetchurl and build up the url using finalAttrs.

I initially fetched the git repo after pondering libxz and how the tarball doesn't have to match the source. But I imagine that's a nixpkgs wide change.

Reply:
Not sure I fully understand the last sentence of your previous comment, @quag, would you mind elaborating a bit?

Reply:
The libxz attack changed how the tarballs were built. Distros that pulled from the git repo rather than the tarball were immune.

From [https://www.akamai.com/blog/security-research/critical-linux-backdoor-xz-utils-discovered-what-to-know](https://www.akamai.com/blog/security-research/critical-linux-backdoor-xz-utils-discovered-what-to-know):
> The backdoor is quite complex. For starters, you won’t find it in the xz GitHub repository (which is currently disabled, but that’s besides the point). In what seems like an attempt to avoid detection, instead of pushing parts of the backdoor to the public git repository, the malicious maintainer only included it in source code tarball releases.

While adding gema I thought of libxz and automatically used the git repo in preference to the uploaded tarballs. You rightly point out that there is a performance cost to using git over the tarball.